### PR TITLE
Fix occasional WebView2 Crash when restoring minimized app (0x80070057)

### PR DESCRIPTION
**Pull request by DmitriyKomin**
If an app is minimized while WebView2 has focus, and is then restored, it occasionally crashes with 0x80070057/ERROR_INVALID_PARAMETER. In my local repro, i observe this on ~1 out of 20 tries. In such cases the Windows restore operation is initiated but for some reason not completed; the app remains minimized. Nonetheless Xaml internal focus handling proceeds to restore focus to WebVIew2, which in turn calls CoreWebView2Controller::MoveFocus to restore web element focus. The latter call fails in Win32 ::SetFocus() API w/ ERROR_INVALID_PARAMETER; this is expected since the target HWND remained minimized.

The fix is to ignore HR 0x80070057 in this situation. Since the app remained minimized in the repro case, there is no observable effect and no additional state cleanup is required. The next attempt to restore the app should restore focus correctly.

### CODE REVIEW

File: dev/WebView2/WebView2.cpp [hunk 1]
```diff
@@ -1255,14 +1255,25 @@ void WebView2::FireCoreWebView2Initialized(winrt::hresult exception)
     m_coreWebView2InitializedEventSource(*this, *eventArgs);
 }
 
-void WebView2::HandleGotFocus(const winrt::Windows::Foundation::IInspectable&, const winrt::RoutedEventArgs&) noexcept
+void WebView2::HandleGotFocus(const winrt::Windows::Foundation::IInspectable&, const winrt::RoutedEventArgs&)
 {
     if (m_coreWebView && m_xamlFocusChangeInfo.m_isPending)
     {
         // In current CoreWebView2 API, MoveFocus is not expected to fail, so set m_webHasFocus here rather than
```
Comment (krschau) [inline dev/WebView2/WebView2.cpp:1262]: Is this comment now incorrect? Should the "m_webHasFocus = true" move after MoveFocus in the try block?

File: dev/WebView2/WebView2.cpp [hunk 2]
```diff
@@ -1255,14 +1255,25 @@ void WebView2::FireCoreWebView2Initialized(winrt::hresult exception)
...
         // in asynchronous (MOJO/cross-proc) CoreWebView2 GotFocus event.
         m_webHasFocus = true;
-        m_coreWebViewController.MoveFocus(m_xamlFocusChangeInfo.m_storedMoveFocusReason);
+        try
+        {
+            m_coreWebViewController.MoveFocus(m_xamlFocusChangeInfo.m_storedMoveFocusReason);
+        }
+        catch (winrt::hresult_error e)
+        {
+            if (e.code().value != E_INVALIDARG)
```
Comment (krschau) [inline dev/WebView2/WebView2.cpp:1271]: Please add comment about what's happening here.

**Review by krschau — CHANGES_REQUESTED** [review]

File: dev/WebView2/WebView2.cpp (hunk 2 above)
Comment (DmitriyKomin) [reply dev/WebView2/WebView2.cpp:1275]: Sorry, I had a comment written but realized I didn't commit the change... fixed now :-)

File: dev/WebView2/WebView2.cpp (hunk 1 above)
Comment (DmitriyKomin) [reply dev/WebView2/WebView2.cpp]: Yep, comment mo longer applies - fixed (also forgot to push this). 

RE m_webHasFocus=true, good suggestion, thanks :-) made this change also.

**Review by krschau — APPROVED** [review]

**Pull request merged**